### PR TITLE
[spec/interpreter] Correct and test segment offset syntax

### PR DESCRIPTION
**Pull request by rossberg**
Addresses #662 and adds tests.

### CODE REVIEW

**Comment by rossberg** [comment]
@binji, friendly ping requesting review.

File: test/core/data.wast [hunk 1]
```diff
@@ -0,0 +1,110 @@
+;; Test the data section
+
+;; Syntax
+
+(module
```
Comment (binji) [inline test/core/data.wast]: Some of these tests seem to be duplicated in `memory.wast`, is it worth moving these tests there or removing those instead?

Comment (rossberg) [reply test/core/data.wast]: Good point. I moved all data-specific tests over and added quite a few more. PTAL

File: test/core/data.wast [hunk 2]
```diff
@@ -3,79 +3,226 @@
...
+  (data (i32.const 1) "b")
+  (data (i32.const 2) "cde")
+  (data (i32.const 3) "f")
+  (data (i32.const 2) "g")
+  (data (i32.const 1) "h")
+)
 
 (module
-  (memory 2)
+  (global (import "spectest" "global") i32)
```
Comment (binji) [inline test/core/data.wast]: Needs to be `global_i32` now, right?

Comment (rossberg) [reply test/core/data.wast]: Rebased and adjusted.

**Pull request merged**